File: Leetcode/0001-two-sum/solution.py
class Solution(object):
    def twoSum(self, nums, target):
       
        # Dictionary to store value to index mappings
        dictionary = {}
        
        # Iterate over the array
        for idx, element in enumerate(nums):
            # Calculate the complement
            complement = target - element
            
            # Check if the complement is already in the dictionary
            if complement in dictionary:
                return [dictionary[complement], idx]
            
            # If complement is not found, store the current element with its index
            dictionary[element] = idx

File: Leetcode/0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> complement = new HashMap<>();
        complement.put(target-nums[0],0);
        for(int i=1;i<nums.length;i++){
            if(complement.containsKey(nums[i]))
                return new int[] {complement.get(nums[i]), i};
            else
                complement.put(target-nums[i],i);
        }
        return new int[2];
    }
}
